### PR TITLE
Correctly read phrase frequencies for sorting

### PR DESCRIPTION
**Pull request by yan12125**
Closes https://github.com/chewing/libchewing/issues/334

### CODE REVIEW

**Comment by yan12125** [comment]
Well, no CI tests are run as travis-ci.org is ceased. With this change, all tests are still green on my machine.

**Comment by ShikiSuen** [comment]
@yan12125 What about GitHub continuous integration?

File: src/tools/init_database.c [hunk 1]
```diff
@@ -531,7 +531,7 @@ void insert_leaf(NODE * parent, long phr_pos, uint32_t freq)
     NODE *pnew;
 
     for (p = parent->pFirstChild; p && GetUint16(p->data.key) == 0; prev = p, p = p->pNextSibling)
-        if (GetUint16(p->data.phrase.freq) <= freq)
```
Comment (ShikiSuen) [inline src/tools/init_database.c:534]: 不是太懂 C，不知道這裡是否可以用 GetUint32。

Comment (yan12125) [reply src/tools/init_database.c:534]: 如果要用uint32，詞庫的格式以及libchewing要跟著改，用四個uint8_t表示頻率。目前我是以相容現有的libchewing為目標。

**Comment by yan12125** [comment]
> What about GitHub continuous integration?

That's a good idea. If libchewing core developers are fine with GitHub Actions, I can try it in another PR.

**Comment by yan12125** [comment]
Rebased to test the change through GitHub Actions, which was added recently in https://github.com/chewing/libchewing/pull/349.

**Comment by yan12125** [comment]
@kanru Mind to have another look?

Per https://github.com/chewing/libchewing/issues/334#issuecomment-1133582837, it was expected that `init_database.c` would be removed in favor of `init_database.rs`. It turns out that the latter uses a different file format - 32 bit integers for phrase frequency [1] rather than 24-bit integers as in `init_database.c`. As long as `WITH_RUST=OFF` is supported, `init_database.c` should be maintained. Thus, I think it's time to revisit this fix.

[1] https://github.com/chewing/libchewing/blob/cebaeb3874ba54ad173f6750245235e02ce20d8f/tools/src/bin/init_database.rs#L92

**Review by kanru — APPROVED** [review]

**Pull request merged**